mirgecom/euler.py

```diff
@@ -20,17 +22,238 @@
 THE SOFTWARE.
 """
 
-# import numpy as np
-# import numpy.linalg as la  # noqa
-# from pytools.obj_array import (
-#         join_fields, make_obj_array,
-#         with_object_array_or_scalar)
-# from meshmode.mesh import BTAG_ALL, BTAG_NONE  # noqa
-# # TODO: Remove grudge dependence?
-# from grudge.eager import with_queue
-# from grudge.symbolic.primitives import TracePair
+import numpy as np
+import numpy.linalg as la  # noqa
+from pytools.obj_array import (
+    join_fields,
+    make_obj_array,
+    with_object_array_or_scalar,
+)
+import pyopencl.clmath as clmath
+from meshmode.mesh import BTAG_ALL, BTAG_NONE  # noqa
+from mirgecom.boundary import BoundaryBoss
 
+# TODO: Remove grudge dependence?
+from grudge.eager import with_queue
+from grudge.symbolic.primitives import TracePair
 
-# __doc__ = """
-# .. autofunction:: euler_operator
-# """
+
+__doc__ = """
+.. autofunction:: inviscid_operator
+"""
+
+
+#
+# Euler flow eqns:
+# d_t(q) + nabla .dot. f = 0 (no sources atm)
+# state vector q: [rho rhoE rhoV]
+# flux tensor f: [rhoV (rhoE + p)V (rhoV.x.V + delta_ij*p)]
+#
+
+
+def _interior_trace_pair(discr, vec):
+    i = discr.interp("vol", "int_faces", vec)
+    e = with_object_array_or_scalar(
+        lambda el: discr.opposite_face_connection()(el.queue, el), i
+    )
+    return TracePair("int_faces", i, e)
+
+
+def _inviscid_flux(discr, q):
+
+    # q = [ rho rhoE rhoV ]
+    ndim = discr.dim
+
+    rho = q[0]
+    rhoE = q[1]
+    rhoV = q[2:]
+
+    # --- EOS stuff TBD ---
```

Comment:
Shouldn't this be using the EOS?

Reply:
Yes?  I think you may have been referring to a comment that was removed in a previous commit.